shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt

```diff
@@ -102,6 +104,8 @@ object JsonLdUtils {
                 throw BadRequestDataException("Unable to parse input payload")
             expandedList[0]
         } catch (e: Exception) {
+            if (e is JsonParseException)
```

Comment:
I added this check since the try block throws JsonParse exceptions and JsonLdParse exceptions.
- For JsonParseException, we throw the InvalidRequestException
- Else, we throw BadRequestDataException